### PR TITLE
Proof of concept

### PR DESCRIPTION
**Pull request by novemberborn**
@jamestalmage 👋

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const { join } = require('path')
+const { Script } = require('vm')
+
+exports.bundleDependencies = function (mid) {
+  const execa = require('execa')
+  return execa.stdout(
+    process.execPath,
```
Comment (jamestalmage) [inline index.js:7]: Forking in case some modules we want to cache are already loaded?

File: bundle-dependencies.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+recording = false
+
+// Load further dependencies after recording the files that are to be bundled.
+const Module = require('module')
+const { dirname } = require('path')
+const { transform } = require('babel-core')
+const wrapListener = require('babel-plugin-detective/wrap-listener')
+const resolveFrom = require('resolve-from')
+
+const rewriteRequire = wrapListener((path, file) => {
```
Comment (jamestalmage) [inline bundle-dependencies.js:24]: I wonder if just using [`detective`](https://www.npmjs.com/package/detective) would be more performant than loading up all of babel. `babel-detective` is beneficial when we are using babel to perform other transforms, not sure how it compares performance-wise to the original.

**Review by jamestalmage — COMMENTED** [review]
Interesting so far.

How were you measuring your performance?

`bundleDependencies('babel-core')` takes a lot longer than just `require('babel-core')` on my machine.

**Comment by jamestalmage** [comment]
see #2.

I think this is promising enough to warrant further exploration.

**Comment by novemberborn** [comment]
> How were you measuring your performance?

``` js
const convert = require('convert-hrtime')
const start = process.hrtime()

require('babel-core')

console.log(convert(process.hrtime(start)))
```

File: index.js (hunk 1 above)
Comment (novemberborn) [reply index.js:7]: Yup.

File: bundle-dependencies.js (hunk 2 above)
Comment (novemberborn) [reply bundle-dependencies.js:24]: Maybe, but I need to rewrite the `require` calls to use absolute paths. That way the module lookup on load is a lot easier.

**Pull request merged**